packages/react/src/internal/components/Announce.tsx

```diff
@@ -0,0 +1,72 @@
+import '@primer/live-region-element/define'
+import {announceFromElement} from '@primer/live-region-element'
+import React, {useEffect, useRef, type ElementRef} from 'react'
+import Box from '../../Box'
+import {useEffectOnce} from '../hooks/useEffectOnce'
+
+export type AnnounceProps = React.ComponentPropsWithoutRef<typeof Box> & {
+  /**
+   * The politeness level to use for the announcement
+   * @default polite
+   */
+  politeness?: 'polite' | 'assertive'
+}
+
+/**
+ * `Announce` is a component that will announce the text content of the
+ * `children` passed in to screen readers using the given politeness level. It
+ * will also announce any changes to the text content of `children`
+ */
+export function Announce({children, politeness = 'polite', ...rest}: AnnounceProps) {
+  const ref = useRef<ElementRef<'div'>>(null)
+  const savedPoliteness = useRef(politeness)
+
+  useEffect(() => {
+    savedPoliteness.current = politeness
+  }, [politeness])
```

Comment:
(non blocking) Why do we save this in a ref, but also update it whenever it changes?

Is this the equivalent of using it from props or do we intentionally only want to use it after an effect lifecycle?

Reply:
Ah I see!